parser/parser.go

```diff
@@ -0,0 +1,161 @@
+// go run parser.go [file_path] [semster]
+// example: go run parser.go "Fall 2019.csv" 19F
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"parser/configs"
+	"reflect"
+
+	"os"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+type Class struct {
+	subject           string
+	catalogNumber     string
+	section           string
+	gradeDistribution bson.A
+}
+
+func main() {
+	URI := configs.EnvMongoURI()
+	if len(os.Args) != 3 {
+		fmt.Println("usage: \ngo build parser.go\n./parser.exe [file_path] [academic_session]\nexample: \ngo build parser.go\n./parser.exe \"Fall 2019.csv\" 19F")
+	}
+	filePath := os.Args[1]
+	academicSession := os.Args[2]
+	var classes []Class
+	file, err := os.Open(filePath)
+	if err != nil {
+		fmt.Println("could not open file " + filePath)
+		fmt.Println(err)
+		return
+	}
+	reader := csv.NewReader(file)
+	records, err := reader.ReadAll() // records is [][]strings
```

Comment:
How big is the grades csv file? I worry that reading all of it may cause excessive memory use. It might be better to load one record into memory at a time with reader.Read(). If the csv file isn't that big, then what you have here is good.
https://pkg.go.dev/encoding/csv#Reader.Read